local/network.go

```diff
@@ -563,18 +566,24 @@ func (ln *localNetwork) Healthy(ctx context.Context) error {
 	errGr, ctx := errgroup.WithContext(ctx)
 	for _, node := range ln.nodes {
 		node := node
+		nodeName := node.GetName()
 		errGr.Go(func() error {
 			// Every [healthCheckFreq], query node for health status.
 			// Do this until ctx timeout or network closed.
 			for {
+				if node.Status() != status.Running {
+					// If we had stopped this node ourselves, it wouldn't be in [ln.nodes].
+					// Since it is, it means the node stopped unexpectedly.
```

Comment:
And should we remove it from `ln.nodes` now or is it not necessary?

Reply:
Currently it seems not necessary, we are doing the less possible recovery, as it is not clear which action may make the network stable again. A user shall decide what to do based on he is observing. He may call RemoveNode if he finds out that the network can recover from the problem.